x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -647,48 +625,64 @@ public void setFrequency(TimeValue frequency) {
             this.frequency = frequency;
         }
 
-        public void setParsedQuery(QueryBuilder query) {
+        public void setQuery(Map<String, Object> query) {
+            setQuery(query, true);
+        }
+
+        public void setQuery(Map<String, Object> query, boolean lenient) {
+            this.query = ExceptionsHelper.requireNonNull(query, QUERY.getPreferredName());
```

Comment:
What happens if an empty map is supplied?  Does that fail the validation below?  If not we should block it here, as it's against the spirit of defaulting to `match_all` that someone can specify an empty query.  (If it already fails the validation below then that's fine - no changes needed here.)

Reply:
Will double check